Kuadrant/authorino
# Adds the possibility to set dynamic response body to denied requests

This PR addresses the issue https://github.com/Kuadrant/authorino/issues/220 to set custom response message in the body of denied requests.

## Verification steps

1) Install and deploy an instance of authorino by running 
```
$ make local-setup
```
2) Apply an Authconfig CR
```
$ kubectl apply -n authorino -f https://raw.githubusercontent.com/kuadrant/authorino-examples/main/hello-world/authconfig.yaml
```
3) Add a denyWith policy to the Authconfig CR
```
spec:
...
  denyWith:
    unauthenticated:
      code: 302
      headers:
      - name: Location
        valueFrom:
          authJSON: http://matrix-quotes-authorino.127.0.0.1.nip.io:8000/login.html?redirect_to={context.request.http.path}
      message:
        valueFrom:
          authJSON: test{context.request.http.path}
```
4) Make a request against the talker API and check the response body `x-ext-auth-reason` has the custom message `test/hello`
```
$ curl http://talker-api-authorino.127.0.0.1.nip.io:8000/hello -i

HTTP/1.1 302 Found
location: http://matrix-quotes-authorino.127.0.0.1.nip.io:8000/login.html?redirect_to=/hello
x-ext-auth-reason: test/hello
date: Wed, 06 Apr 2022 10:48:27 GMT
server: envoy
content-length: 0
```
